### ISSUE TITLE
Hello

### ISSUE THREAD

**Issue by mrcodegit** [danielterwiel/test_application#1]
Hi Daniel,

Thank you for taking the time to complete our test and to agree to provide some improvements.

It is a fundamental part of the interview process. There are no time constraints from my side, so that you can provide fixes and discussion on your own time, at your own pace.

Cheers, Maciej!

**Comment by danielterwiel** [comment]
Hi @mrcodegit !

Thanks for giving me the extra time I need to finish the test. The reason I submitted it without the nice-to-haves was because I wanted to respect the given timeframe of 2-4 hours. The initial time frame made me want to rush things hence the somewhat lousy quality of the deliverable. I can provide better quality when I get the time to think about an implementation. It has taken me longer than I expected: I had some errands to run this weekend and I ran into some pesky blockers.

To show that I provide better quality when I'm not under as much pressure I decided to go the extra mile and persist the pagination and search state as URLSearchParams to provide a better user experience when refreshing and allow sharing of URL's. I discussed this with Thomasz so I figured I had to put my money where my mouth is. I could still iterate on this, but I think this should provide a clear view of my React knowledge.

My GraphQL experience is very minimal and testing it was somewhat challenging at first. I'm looking forward to all your feedback.
